### PR TITLE
[Android] Password color

### LINKED ISSUE #4435: [TextInput] Setting secureTextEntry={true} forces input text to color:#000
There doesn't seem to be anything I can do to get it to the set style color. This only happens when secureTextEntry is true.

### PR DESCRIPTION
**Pull request by sooth-sayer**
Closes #4435

### CODE REVIEW

**Comment by mkonicek** [comment]
Thanks for the PR @sooth-sayer. What's your test plan? Have you tried setting this from JS and running the code? Can you add a screenshot showing how this looks?

Does the feature work on iOS exactly the same way? Asking because consistency between iOS and Android APIs where possible is great.

**Comment by sooth-sayer** [comment]
ios colored password
android colored password mp4
@mkonicek I add colred password textinput to examples.

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java [hunk 1]
```diff
@@ -165,6 +165,11 @@ public void updateExtraData(ReactEditText view, Object extraData) {
     }
   }
 
+  @ReactProp(name = ViewProps.COLOR, defaultInt = 0)
+  public void setColor(ReactEditText view, @Nullable Integer color) {
```
Comment (dmmiller) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: You should probably say that this has customType="color" as well.

I'm not sure if defaultInt will keep null from actually showing up.  If null does show up or defaultInt you probably need to set it back to whatever it was before.

**Comment by caphun** [comment]
any plans for this fix to be merged in soon? we tested on a forked version of 0.20 and it worked well.

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java (hunk 1 above)
Comment (sooth-sayer) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: @dmmiller Could you explain in more detail about defaultInt?

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java [hunk 2]
```diff
@@ -165,6 +165,11 @@ public void updateExtraData(ReactEditText view, Object extraData) {
     }
   }
 
+  @ReactProp(name = ViewProps.COLOR, defaultInt = 0, customType="color")
```
Comment (dmmiller) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: Can you remove the defaultInt and check one thing for me?  If you set the color to something and then you rerender with no color set, does it revert back to the proper color?

**Comment by dmmiller** [comment]
One thing to check which I put inline and then I will accept.  Thanks

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java (hunk 2 above)
Comment (sooth-sayer) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: @dmmiller If i set the color and then rerender without color, it fails with "Error while updating property 'color' of a view managed by: AndroidTextInput"

Comment (dmmiller) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: If you leave the defaultInt bit in, does it then work?

Comment (sooth-sayer) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: No. it fails regardless of defaultInt presence

Comment (dmmiller) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java]: We need to figure that out before we can take it.  It's possible for people to want to change their color and then back.  You could also look at why the color that works via spans doesn't for password colors.

**Comment by dmmiller** [comment]
@sooth-sayer I'd like to take this, but I want to know the answer to whether changing colors and removing colors correctly updates.  Can you please indicate that?

**Comment by sooth-sayer** [comment]
@dmmiller No, it fails when I remove color. I'll try to fix it as soon as I will have enough time.

**Comment by janicduplessis** [comment]
You can probably remove the defaultInt, check for null and use the default text color from the resources with something like 
```
getResources().getColor(android.R.color.primary_text_dark);
```

**Comment by janicduplessis** [comment]
Just noticed there's another PR that does this that landed already #6563. I guess we can close this one.

**Comment by mkonicek** [comment]
Thanks @janicduplessis!

**Pull request closed without merging**